### ISSUE TITLE
load model assertion failed on visual studio vcpkg

### ISSUE THREAD

**Issue by ericwang10** [Dobiasd/frugally-deep#370]
Hi Dobiasd,

This is my first time working with C++ for a school project. 

I downloaded my trained keras model and converted it to a fdeep_model.json using the python convert model file. However, when I'm trying to load the file in c++ visual studio, I get an assertion failed error from fplus. The strange thing is that the model loading worked successfully on my teammate's computer. 

Would you mind running it on your end to see if you have an error? I've been having lots of issues installing all the prerequisite packages with cmake so I'm wondering if this is an issue due to my package installations? 

My tensorflow version is 2.10 and python version is 3.10.6

I've attached some screenshots below as well as the keras model.h5 and json file 

![image](https://user-images.githubusercontent.com/67357603/204118790-fafa7ebd-af65-4157-94de-49dba2bd805e.png)

![image](https://user-images.githubusercontent.com/67357603/204118786-ae2558e6-6875-4c0d-834d-e65e32e8d2d9.png)

[model.zip](https://github.com/Dobiasd/frugally-deep/files/10097500/model.zip)

**Comment by Dobiasd** [comment]
Hi, and thanks for the good explanation of your problem. :+1:

I've created `Dockerfile` to test your model:

```Dockerfile
FROM python:3.10.6

RUN apt-get update
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get install -y build-essential cmake

RUN pip3 install tensorflow==2.10.0 keras==2.10.0

RUN apt-get remove --purge -y cmake
RUN pip install cmake --upgrade

RUN git clone -b 'v0.2.18-p0' --single-branch --depth 1 https://github.com/Dobiasd/FunctionalPlus && cd FunctionalPlus && mkdir -p build && cd build && cmake .. && make && make install
RUN git clone -b '3.4.0' --single-branch --depth 1 https://gitlab.com/libeigen/eigen.git && cd eigen && mkdir -p build && cd build && cmake .. && make && make install && ln -s /usr/local/include/eigen3/Eigen /usr/local/include/Eigen
RUN git clone -b 'v3.10.5' --single-branch --depth 1 https://github.com/nlohmann/json && cd json && mkdir -p build && cd build && cmake -DJSON_BuildTests=OFF .. && make && make install
RUN git clone -b 'v0.15.19-p0' --single-branch --depth 1 https://github.com/Dobiasd/frugally-deep && cd frugally-deep && mkdir -p build && cd build && cmake .. && make && make install

WORKDIR /frugally-deep

RUN wget https://github.com/Dobiasd/frugally-deep/files/10097500/model.zip
RUN unzip model.zip

RUN CUDA_VISIBLE_DEVICES='' taskset --cpu-list 1 python3 ./keras_export/convert_model.py keras_model.h5 fdeep_model.json

RUN echo '#include "fdeep/fdeep.hpp"\n\
int main()\n\
{\n\
    const auto model = fdeep::load_model("fdeep_model.json");\n\
}' >> main.cpp

RUN g++ -std=c++14 -O3 -march=native main.cpp -o main

RUN ./main
```

It works:

```
Loading json ... done. elapsed time: 0.016621 s
Building model ... done. elapsed time: 0.017323 s
Running test 1 of 1 ... done. elapsed time: 0.001938 s
Loading, constructing, testing of fdeep_model.json took 0.036293 s overall.
```

So in general, frugally-deep is fine with loading your model in C++.

To continue, I recommend:
- Make sure you have the latest version of FunctionalPlus installed.
- Make sure you have the latest version of frugally-deep installed.
- Try printing not only the exception but also the stack trace (or run in Debug mode), so we can see where the failing `fplus::maybe::is_just()` call is coming from.

**Comment by ericwang10** [comment]
Hi Dobias,

Thank you for the quick response and the tips! Your advice to check the stack trace helped me find the issue. 

I think the issue is due to a layer in my model. I added a rescaling layer 
`model = Sequential()`
`model.add(layers.Rescaling(1./255, input_shape = (48, 48, 1)))`
to normalize pixel values between 0 and 1. However, I think fdeep doesn't support this layer. 
![image](https://user-images.githubusercontent.com/67357603/204233585-2bbb63a2-08eb-4f17-88a5-10fa0ce2c5db.png)

I retrained the model in python without the rescaling layer and was able to successfully load the model with fdeep! 

However, it's strange that under different computer settings, this runtime error doesn't occur and the fdeep model loads with the rescaling layer. My groupmates loaded the model successfully using command line cmake on windows and macOS. It also seemed that you didn't have any issues loading the json model. Is this issue caused by how Visual Studio handles the run time error? 

In the future, should we stick to the built-in normalization parameters in `fdeep::tensor_from_bytes`?

Thanks!

**Comment by Dobiasd** [comment]
frugally-deep *does* support `Rescaling` layers. It was [added around 7 months ago](https://github.com/Dobiasd/frugally-deep/commit/f8c09a3af2575543722edc4844abfb649986d4a9).

> Is this issue caused by how Visual Studio handles the run time error?
> In the future, should we stick to the built-in normalization parameters in `fdeep::tensor_from_bytes`?

No, and now. :wink: I think you are simply using an outdated version of frugally-deep.

**Comment by ericwang10** [comment]
Ahhh that makes sense. I see that the vcpkg version is [0.15.10-p0] while the current version is [v0.15.19-p0]. Is there any way I could get the updated version through vcpkg?

**Comment by Dobiasd** [comment]
Oh, I see. I don't yet know how to push new version to vcpkg, but I've just asked [here](https://github.com/Dobiasd/frugally-deep/pull/277#issuecomment-1329116626).

**Comment by Dobiasd** [comment]
@ericwang10 frugally-deep `v0.15.19-p0` is now available on vcpkg. :tada:

**Comment by ericwang10** [comment]
@Dobiasd Thanks a lot! Your package really helped me out a lot with my school project!
